syslog: T6989: convert old configuration format to "advanced"

c-po opened the pull request:
<!-- All PR should follow this template to allow a clean and transparent review -->
<!-- Text placed between these delimiters is considered a comment and is not rendered -->

## Change summary
<!--- Provide a general summary of your changes in the Title above -->

We are still using the legacy config format for rsyslog in https://github.com/vyos/vyos-1x/blob/bb4611544ff73a979f927a7a1c8ea3dd1c56fb48/data/templates/rsyslog/rsyslog.conf.j2

It has been superseded by a new format:

https://www.rsyslog.com/doc/configuration/converting_to_new_format.html

There are some associated issues. For example, we use `$markMessagePeriod` for `set system syslog global marker interval`, which sends mark messages unconditionally. New rsyslog versions support sending mark messages only if there are no real messages, the old directive is now listed as obsolete (https://www.rsyslog.com/doc/configuration/input_directives/rsconf1_markmessageperiod.html) and its replacement seems to be only available in the advanced format.

*  Drop `file` CLI option to log to dedicated files.
*  Drop `user` CLI option to always log to user terminal - should be an op-mode command instead
* Rename `host <ipv4|ipv6|fqdn>` -> `remote <ipv4|ipv6|fqdn>`
* Add new source-address option for outgoing connections under `remote source-address <ipv4|ipv6|fqdn>`
* VRF can be defined per remote as  `remote vrf <name>`
* Move `global preserve-fqdn` -> `preserve-fqdn` - move up one CLI level
* Move `global marker interval` -> `marker interval` - move up one CLI level
* Rename `global` -> `local` as every facility configured under global is actually logged locally to `/var/log/messages`


## Types of changes
<!---
What types of changes does your code introduce? Put an 'x' in all the boxes that apply.
NOTE: Markdown requires no leading or trailing whitespace inside the [ ] for checking
the box, please use [x]
-->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)
<!-- optional: Link to related other tasks on Phabricator. -->
* https://vyos.dev/T6989

## Related PR(s)
<!-- Link here any PRs in other repositories that are required by this PR -->
* https://github.com/vyos/vyos-documentation/pull/1589
* https://github.com/vyos/vyos-build/pull/898

## How to test / Smoketest result

```
vyos@vyos:~$ /usr/libexec/vyos/tests/smoke/cli/test_system_syslog.py
test_console (__main__.TestRSYSLOGService.test_console) ... ok
test_global (__main__.TestRSYSLOGService.test_global) ... ok
test_remote (__main__.TestRSYSLOGService.test_remote) ... ok
test_vrf_source_address (__main__.TestRSYSLOGService.test_vrf_source_address) ... ok

----------------------------------------------------------------------
Ran 4 tests in 9.964s

OK
```

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
<!--- The entire development process is outlined here: https://docs.vyos.io/en/latest/contributing/development.html -->
- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [x] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [x] My change requires a change to the documentation
- [x] I have updated the documentation accordingly

c-po committed:
syslog: T6989: remove "file" logging destination

c-po committed:
syslog: T6989: remove CLI option for user terminal logging

Should be added as runtime option similar to "terminal monitor" known from
other vendors.

c-po committed:
syslog: T6989: rename "host" to "remote"

c-po committed:
syslog: T6989: convert old configuration format to "advanced"

c-po committed:
syslog: T6989: add new source-address CLI option for outgoing connections

c-po committed:
syslog: T6989: move up "global preserve-fqdn" one level in CLI

Move "global preserve-fqdn" one CLI level up, as it relates to all logging
targets (console, global and remote).

c-po committed:
syslog: T6989: move up "global marker" one level in CLI

c-po committed:
syslog: T6989: remove RepeatedMsgReduction configuration option

This is a feature that worked decades ago when logs were small and reviewed by
a human, it fails badly on high volume logs processed by tools.

https://www.rsyslog.com/doc/configuration/action/rsconf1_repeatedmsgreduction.html

c-po committed:
syslog: T6989: add possibility to define VRF per remote

Rsyslog supports individual VRFs per omfwd remote entry - so we should support
this, too.

dmbaturin commented on data/templates/rsyslog/rsyslog.conf.j2:
```diff
@@ -1,78 +1,123 @@
+    action(
+        type="omfile"
+        file="/var/log/messages"
+        queue.size="262144"
```
Can we include some comment about this number? If I see a non-round number, I always start wondering if it's either completely random (and can be arbitrary) or it was calculated in some complex and precise way (and dangerous to change).

c-po replied:
It's a relict from our history:

https://github.com/vyos/vyos-1x/blob/crux/src/conf_mode/syslog.py#L132

We probably should rethink this behavior and decide for a new/the same number and update it with a proper comment.

c-po replied:
Adjusted

c-po committed:
syslog: T6989: increase local log size form 256KiB -> 1MiB

c-po committed:
syslog: T6989: rename "global" to "local"

The previously "global" options actually were only relevant for the local
logging to /var/log/messages.

jestabro commented:
Note that the smoketest `test_console` raises errors, although it passes as unittest.

c-po committed:
smoketest: T6989: extend cli_set() signature to pass CLI node value

c-po committed:
init: T6989: start syslog early

c-po committed:
defaults: T6989: provide single source of systemd services

Some systemd services are re-used over multiple configuration files. Keep a
single source of the real systemd names and only reference them by dictionary
keys.

c-po commented:
> Note that the smoketest `test_console` raises errors, although it passes as unittest.

Thank you - issue fixed in commit 9212b401d33

Apachez- commented:
It looks like this commit wasnt properly tested before pushed to the public tree according to:

https://github.com/vyos/vyos-nightly-build/actions/runs/13210103854/job/36882252374#step:5:2710

https://github.com/vyos/vyos-1x/blame/38dacfeb8fb66d135f8ad219ff3c2fc7504f3029/smoketest/scripts/cli/test_system_syslog.py#L62

Above seems the cause for why there have been no succeeded nightly builds since 3th feb (today is 8th feb):

https://github.com/vyos/vyos-nightly-build/releases

https://github.com/vyos/vyos-nightly-build/actions
